File: AppointmentServiceTest.java
// Author Name: Owen Lovell

// Date: 10/06/2024

// Course ID: CS-320-13376-M01

// Description: This is the appointment service test class that adds or deletes an appointment


package module5;

import static org.junit.jupiter.api.Assertions.*;

import java.util.Date;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;

class AppointmentServiceTest {

	@Test
	@DisplayName("Test Add Appointment")
	@Order(1)
	void testAddAppointment() {
		AppointmentService service = new AppointmentService();
		service.addAppointment("10/25/2024", "The doctor is available");
		
		Date testDate = new Date(10/25/2024);
		if(service.getAppointment("0").getAppointmentDate().compareTo(testDate) > 0 || service.getAppointment("0").getAppointmentDate().compareTo(testDate) < 0) {
			System.out.println("The Date was stored as Appointment object as: " + service.getAppointment("0").getAppointmentDate());
			fail("Date doesn't match the string, something went wrong creating the date");
		}
		assertTrue(service.getAppointment("0").getDescription().equals("The doctor is available"));
	}

	@Test
	@DisplayName("Test Delete Appointment")
	@Order(2)
	void testDeleteAppointment() {
		AppointmentService service = new AppointmentService();
		service.addAppointment("10/24/2024", "Help is on the way");
		service.addAppointment("10/26/2024", "Routine check up");
		
		int listSizeBefore = service.appointmentList.size();
		System.out.println("List size 1 =" + listSizeBefore);
		service.deleteAppointment("1");
		int listSizeAfter = service.appointmentList.size();
		System.out.println("List size 2 =" + listSizeAfter);
		
		assertTrue(listSizeBefore > listSizeAfter);

	}

}

File: Projects/Enhancement-2/Enhanced/appointment_service_test.py
# Author: Owen Lovell
# Date Updated: 12/13/2025
# Course ID: CS-499-17242-M01

import unittest
from appointment_service import *


class TestAppointmentService(unittest.TestCase):
    # Test to check creation and deletion of appointment
    def test_appointment_create(self):
        test_instance = AppointmentList()
        test_instance.add_appointment("1/1/2025", "testing functionality of service")
        self.assertTrue(test_instance.get_appointment("1/1/2025"))
        test_instance.remove_appointment("1/1/2025")
        self.assertFalse(test_instance.get_appointment("1/1/2025"))


if __name__ == '__main__':
    unittest.main()
